sycl/doc/extensions/accessor_properties/SYCL_INTEL_buffer_location.asciidoc

```diff
@@ -0,0 +1,97 @@
+= SYCL_INTEL_buffer_location
+
+== Introduction
+IMPORTANT: This specification is a draft.
+
+NOTE: Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by permission by Khronos.
+
+NOTE: This document is better viewed when rendered as html with asciidoctor.  GitHub does not render image icons.
+
+This document describes an extension that adds a compile-time-constant property to tell the SYCL device compiler and runtime in which memory a particular accessor should be allocated.  This is useful on targets that possess more than one type of global memory.  
+
+== Name Strings
+
++SYCL_INTEL_buffer_location+
+
+== Contributors
+Joe Garvey, Intel
+
+== Notice
+
+Copyright (c) 2020 Intel Corporation.  All rights reserved.
+
+== Status
+
+Working Draft
+
+This is a preview extension specification, intended to provide early access to a feature for review and community feedback. When the feature matures, this specification may be released as a formal extension.
+
+Because the interfaces defined by this specification are not final and are subject to change they are not intended to be used by shipping software products.
+
+== Version
+
+Built On: {docdate} +
+Revision: A
+
+== Contact
+Michael Kinsner, Intel (michael 'dot' kinsner 'at' intel 'dot' com)
+
+== Dependencies
+
+This extension is written against the SYCL 2020 pre-provisional specification and the SYCL_INTEL_accessor_properties extension.
+
+The use of this extension requires a target that supports SPV_INTEL_fpga_buffer_location if SPIR-V is used as the intermediate representation.  
+
+== Overview
+
+On targets that provide more than one type of global memory, knowing that a particular pointer can only access one of those memory types at compile time can enable compiler optimizations.  
+This extension adds an accessor property to indicate to the runtime in which of these memories the buffer corresponding to this accessor should be allocated and to inform the compiler that all accesses made through that accessor can only ever interact with the given memory.
+This information is not a hint; it is a functional requirement of the program that must be respected.
+
+== Modifications to the SYCL 2020 Pre-Provisional Specification
+
+=== Section 4.7.6.9.2 Device buffer accessor properties
+
+Add a new property and global variable to the listing:
+
+```c++
+namespace sycl {
+...
+namespace INTEL {
```

Comment:
@GarveyJoe have you considered swapping `INTEL` and `property` namespaces in the nesting?
The thing I find confusing in the namespace nesting, is that property namespace is put inside of INTEL (aka hardware extension) namespace, but we will have some other properties, that will have not only FPGA usage, hence these properties shall be placed in ONEAPI (aka language extensions) namespace.

Reply:
I was trying to be consistent with the spirit of the rules from the SYCL 2020 provisional spec which indicates that vendor extensions should be in their own namespace nested in the sycl namespace.  @gmlueck is this the recommended way to nest these namespaces?

Reply:
That makes sense, Thanks.

Reply:
If you are asking whether this extension should be in the `INTEL` vs `ONEAPI` namespace, I think @jbrodman is the right person to answer.

Regarding the order of the namespaces `sycl::INTEL::property` vs. `sycl::property::INTEL`, I think the first is preferable (which is what you have).